spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -0,0 +1,280 @@
+                                Transform[] transforms,
+                                Map<String, String> properties)
+      throws TableAlreadyExistsException {
+    Schema icebergSchema = SparkSchemaUtil.convert(schema);
+    try {
+      return new SparkTable(icebergCatalog.createTable(
+          buildIdentifier(ident),
+          icebergSchema,
+          Spark3Util.toPartitionSpec(icebergSchema, transforms),
+          properties.get("location"),
+          properties));
+    } catch (AlreadyExistsException e) {
+      throw new TableAlreadyExistsException(ident);
+    }
+  }
+
+  @Override
+  public StagedTable stageCreate(Identifier ident, StructType schema, Transform[] transforms,
+                                 Map<String, String> properties) throws TableAlreadyExistsException {
+    Schema icebergSchema = SparkSchemaUtil.convert(schema);
+    try {
+      return new StagedSparkTable(icebergCatalog.newCreateTableTransaction(buildIdentifier(ident), icebergSchema,
+          Spark3Util.toPartitionSpec(icebergSchema, transforms), properties.get("location"), properties));
+    } catch (AlreadyExistsException e) {
+      throw new TableAlreadyExistsException(ident);
+    }
+  }
+
+  @Override
+  public StagedTable stageReplace(Identifier ident, StructType schema, Transform[] transforms,
+                                  Map<String, String> properties) throws NoSuchTableException {
+    Schema icebergSchema = SparkSchemaUtil.convert(schema);
+    try {
+      return new StagedSparkTable(icebergCatalog.newReplaceTableTransaction(buildIdentifier(ident), icebergSchema,
+          Spark3Util.toPartitionSpec(icebergSchema, transforms), properties.get("location"), properties,
+          false /* do not create */));
+    } catch (org.apache.iceberg.exceptions.NoSuchTableException e) {
+      throw new NoSuchTableException(ident);
+    }
+  }
+
+  @Override
+  public StagedTable stageCreateOrReplace(Identifier ident, StructType schema, Transform[] transforms,
+                                          Map<String, String> properties) {
+    Schema icebergSchema = SparkSchemaUtil.convert(schema);
+    return new StagedSparkTable(icebergCatalog.newReplaceTableTransaction(buildIdentifier(ident), icebergSchema,
+        Spark3Util.toPartitionSpec(icebergSchema, transforms), properties.get("location"), properties,
+        true /* create or replace */));
+  }
+
+  @Override
+  public SparkTable alterTable(Identifier ident, TableChange... changes)
+      throws NoSuchTableException {
+    SetProperty setLocation = null;
+    SetProperty setSnapshotId = null;
+    SetProperty pickSnapshotId = null;
+    List<TableChange> propertyChanges = Lists.newArrayList();
+    List<TableChange> schemaChanges = Lists.newArrayList();
+
+    for (TableChange change : changes) {
+      if (change instanceof SetProperty) {
+        SetProperty set = (SetProperty) change;
+        if ("location".equalsIgnoreCase(set.property())) {
```

Comment:
Should we use `TableCatalog.PROP_LOCATION`? What about `comment`, `owner`? I recall some bug fixes in Spark but will need to check again as I forgot the context.

Reply:
I'll double-check these.

Reply:
Updated to use `PROP_LOCATION`. While `comment` and `owner` are reserved properties in Spark, but Iceberg tracks these as table properties.

Side note: we have a class to sync some table properties into other metadata fields in our metastore implementation. But the Iceberg API doesn't explicitly track all of them. I think it makes sense to use table properties here.

Reply:
Makes sense.